core/src/main/java/com/novoda/espresso/ViewTestRule.java

```diff
@@ -1,50 +1,65 @@
 package com.novoda.espresso;
 
 import android.app.Instrumentation;
-import android.content.Intent;
 import android.support.annotation.LayoutRes;
 import android.support.test.InstrumentationRegistry;
 import android.support.test.rule.ActivityTestRule;
 import android.view.View;
+import android.view.ViewGroup;
 
-public class ViewTestRule<T extends View> extends ActivityTestRule<ViewActivity> {
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+
+public class ViewTestRule<T extends View> extends ActivityTestRule<EmptyActivity> {
 
     private final Instrumentation instrumentation;
+    private final ViewCreator<T> viewCreator;
+
+    private T view;
+
+    public ViewTestRule(@LayoutRes int layoutId) {
+        this(new InflateFromXmlViewCreator<T>(layoutId));
+    }
 
-    @LayoutRes
-    private final int id;
+    public ViewTestRule(ViewCreator<T> viewCreator) {
+        this(InstrumentationRegistry.getInstrumentation(), viewCreator);
+    }
 
-    public ViewTestRule(@LayoutRes int id) {
-        super(ViewActivity.class);
-        this.id = id;
-        instrumentation = InstrumentationRegistry.getInstrumentation();
+    public ViewTestRule(Instrumentation instrumentation, ViewCreator<T> viewCreator) {
+        super(EmptyActivity.class);
+        this.instrumentation = instrumentation;
+        this.viewCreator = viewCreator;
     }
 
     @Override
-    protected Intent getActivityIntent() {
-        Intent intent = new Intent(Intent.ACTION_MAIN);
-        intent.putExtra(ViewActivity.EXTRA_LAYOUT_ID, id);
-        return intent;
+    protected void afterActivityLaunched() {
+        super.afterActivityLaunched();
+        final EmptyActivity activity = getActivity();
+        view = viewCreator.createView(activity, (ViewGroup) activity.findViewById(android.R.id.content));
```

Comment:
Is there any good reason to do this outside the main Thread? Thre must be some _fancy_ ways to create a `View` that requires to run on the Main Thread.

Reply:
:+1: can move this so it's on the main thread.

Reply:
commit e8b17519165950d2b7c8731c16bfb11847059480
Author: Ataul Munim <hello@ataulm.com>
Date:   Tue Nov 7 17:21:42 2017 +0000

    Create view on main thread